File: F#/problem2.fs
module problem2

let fib = Seq.unfold (fun (a,b) -> Some( a+b, (b, a+b) ) ) (0,1)
let result = fib 
                |> Seq.takeWhile(fun x -> x < 4000000) 
                |> Seq.filter(fun x -> x % 2 = 0) 
                |> Seq.sum

printfn "%A" result

File: Haskell/problem2.hs
fibs = 0 : 1 : zipWith (+) fibs (tail fibs)

problem2 :: Int
problem2 = 
	sum
	$ filter even
	$ drop 2
	$ takeWhile (\x -> x < 4000000)
	$ fibs

main :: IO()
main = print problem2 